metaci/release/utils.py

```diff
@@ -55,38 +78,131 @@ def update_release_from_github(release, repo_api=None):
     return release
 
 
-def send_release_webhook(project_config, release):
+def send_release_webhook(project_config, release, config_item=None):
     if release is None or not settings.METACI_RELEASE_WEBHOOK_URL:
-        return
+        return  # should we better error handle this?
     logger.info(
         f"Sending release webhook for {release} to {settings.METACI_RELEASE_WEBHOOK_URL}"
     )
     tag = release.git_tag
+
+    steps = []
+    if config_item and settings.METACI_START_STOP_WEBHOOK:
+        implementation_steps = release.implementation_steps.all()
+        for implementation_step in implementation_steps:
+            steps = [
+                implementation_payload(implementation_step.plan.role, config_item)
+                for implementation_step in release.implementation_steps.all()
+            ]
+
     payload = {
         "case_template_id": release.change_case_template.case_template_id,
         "package_name": project_config.project__package__name,
         "version": project_config.get_version_for_tag(tag),
         "release_url": f"{release.repo.url}/releases/tag/{urllib.parse.quote(tag)}",
+        "steps": steps,
     }
-    token = jwt.encode(
-        {
-            "iss": settings.METACI_RELEASE_WEBHOOK_ISSUER,
-            "exp": timegm(datetime.utcnow().utctimetuple()),
-        },
-        settings.METACI_RELEASE_WEBHOOK_AUTH_KEY,
-        algorithm="HS256",
-    )
+    token = jwt_session()
     response = requests.post(
         settings.METACI_RELEASE_WEBHOOK_URL,
         json=payload,
         headers={"Authorization": f"Bearer {token}"},
     )
     result = response.json()
     if result["success"]:
+        result["implementationSteps"] = ["1", "2", "3", "4"]
         with transaction.atomic():
+            if settings.METACI_START_STOP_WEBHOOK:
+                for i in range(0, len(implementation_steps) - 1):
```

Comment:
@davisagli There has to be a better way.

Reply:
How about:

    if "implementation_steps" in result:
        for step_model, step_result in zip(implementation_steps, result["implementationSteps"]):
            step_model.external_id = step_result
            step_model.save()